frontend/src/Components/ProfilePage/ProfilePage.jsx

```
@@ -0,0 +1,231 @@
+      const token = localStorage.getItem("token");
+
+      try {
+        const response = await fetch("http://localhost:3000/api/user/profile", {
+          method: "GET",
+          headers: {
+            "Content-Type": "application/json",
+            Authorization: `Bearer ${token}`,
+          },
+        });
+
+        if (response.ok) {
+          const data = await response.json();
+          setProfile(data);
+          setPinnedServices(data.pinnedServices || []);
+        } else {
+          console.error("Failed to fetch profile data");
+        }
+      } catch (error) {
+        console.error("Error fetching profile data:", error);
+      }
+    };
+
+    fetchProfile();
+  }, []);
+
+  const handleChange = (e) => {
+    setProfile({ ...profile, [e.target.name]: e.target.value });
+  };
+
+  const handleFileChange = (e) => {
+    setSelectedFile(e.target.files[0]);
+  };
+
+  const handleSubmit = async (e) => {
+    e.preventDefault();
+    const token = localStorage.getItem("token");
+
+    try {
+      const formData = new FormData();
+      formData.append("name", profile.name);
+      formData.append("email", profile.email);
+      formData.append("location", profile.location);
+      formData.append("bio", profile.bio);
+      formData.append("interests", profile.interests);
+      if (selectedFile) {
+        formData.append("profilePicture", selectedFile);
+      }
+
+      const response = await fetch("http://localhost:3000/api/user/profile", {
+        method: "PUT",
+        headers: {
+          Authorization: `Bearer ${token}`,
+        },
+        body: formData,
+      });
+
+      if (response.ok) {
+        const updatedProfile = await response.json();
+        setProfile(updatedProfile);
+        setIsModalOpen(false); // Close the modal on successful update
+        alert("Profile updated successfully");
+      } else {
+        console.error("Failed to update profile");
+        alert("Failed to update profile");
```

nit: this error handling is redundant with the one in the `catch` right below. Instead, I'd throw an error here. Something like `if (!response.ok) throw new Error('Response failed with status ${response.status})`. This will cause the code execution to fall through to the `catch`

---

Alright. Done!